### ISSUE TITLE
Crash when selecting certain tables in Realm 0.82.1

### ISSUE THREAD

**Issue by BendyWalker** [uPhyca/stetho-realm#7]
I've added Stetho and this library as in the instructions. When selecting certain tables in the Chrome inspector, they display fine, but others cause a crash with the stack trace below.

The tables that cause a crash and the tables that don't are pretty similar. Both have String IDs and have Realm relationships. It seems (not sure) that the tables that cause a crash have a field that is a number or a date.

```
FATAL EXCEPTION: StethoWorker40
    Process: com.example.example, PID: 13803
    java.lang.IllegalArgumentException: Illegal Argument: ColumnType invalid.
            at io.realm.internal.CheckedRow.nativeGetLong(Native Method)
            at io.realm.internal.UncheckedRow.getLong(UncheckedRow.java:109)
            at com.uphyca.stetho_realm.Database$RowWrapper_0_81.getLink(Database.java:595)
            at com.uphyca.stetho_realm.Database.flattenRows(Database.java:204)
            at com.uphyca.stetho_realm.Database.access$300(Database.java:37)
            at com.uphyca.stetho_realm.Database$1.handleSelect(Database.java:121)
            at com.uphyca.stetho_realm.Database$1.handleSelect(Database.java:101)
            at com.uphyca.stetho_realm.RealmPeerManager.executeSQL(RealmPeerManager.java:123)
            at com.uphyca.stetho_realm.Database.executeSQL(Database.java:100)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.facebook.stetho.inspector.MethodDispatcher$MethodDispatchHelper.invoke(MethodDispatcher.java:96)
            at com.facebook.stetho.inspector.MethodDispatcher.dispatch(MethodDispatcher.java:67)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteRequest(ChromeDevtoolsServer.java:129)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteMessage(ChromeDevtoolsServer.java:111)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.onMessage(ChromeDevtoolsServer.java:87)
            at com.facebook.stetho.websocket.WebSocketSession$1.handleTextFrame(WebSocketSession.java:176)
            at com.facebook.stetho.websocket.WebSocketSession$1.onCompleteFrame(WebSocketSession.java:136)
            at com.facebook.stetho.websocket.ReadHandler.readLoop(ReadHandler.java:46)
            at com.facebook.stetho.websocket.WebSocketSession.handle(WebSocketSession.java:45)
            at com.facebook.stetho.websocket.WebSocketHandler.doUpgrade(WebSocketHandler.java:154)
            at com.facebook.stetho.websocket.WebSocketHandler.handleSecured(WebSocketHandler.java:106)
            at com.facebook.stetho.server.SecureHttpRequestHandler.handle(SecureHttpRequestHandler.java:52)
            at org.apache.http.protocol.HttpService.doService(HttpService.java:248)
            at org.apache.http.protocol.HttpService.handleRequest(HttpService.java:192)
            at com.facebook.stetho.server.LocalSocketHttpServer$WorkerThread.run(LocalSocketHttpServer.java:233)
```

**Comment by pavel-mukhanov** [comment]
we got the same crash on 0.82.0. Some logcat output:

```
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Page.canEmulate
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Console.setTracingBasedTimeline
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Debugger.setAsyncCallStackDepth
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Debugger.skipStackFrames
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Runtime.enable
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Worker.enable
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Timeline.enable
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: IndexedDB.enable
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: Worker.setAutoconnectToWorkers
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: IndexedDB.requestDatabaseNames
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: IndexedDB.requestDatabaseNames
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: IndexedDB.requestDatabaseNames
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: IndexedDB.requestDatabaseNames
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: IndexedDB.requestDatabaseNames
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: ApplicationCache.enable
D/ChromeDevtoolsServer(10963): Method not implemented: Not implemented: ApplicationCache.getFramesWithManifests
D/REALM   (10963): Expected columnType 0, but got 12.
D/REALM   (10963): jni: ThrowingException 3, ColumnType invalid., .
D/REALM   (10963): Exception has been throw: Illegal Argument: ColumnType invalid.
```

**Comment by marcelsanches2** [comment]
I get this error in my app, The error appears when i click in my wrapper object, i not need see this table because only have the references of my objects but if i accidentaly clicking i got the crash, this is the logcat output:

09-12 12:24:08.504  18242-22361/com.pixel.smartticket_mvp E/AndroidRuntime﹕ FATAL EXCEPTION: StethoWorker116
    Process: com.pixel.smartticket_mvp, PID: 18242
    java.lang.IllegalArgumentException: Illegal Argument: ColumnType invalid.
            at io.realm.internal.CheckedRow.nativeGetLong(Native Method)
            at io.realm.internal.UncheckedRow.getLong(UncheckedRow.java:109)
            at com.uphyca.stetho_realm.Database$RowWrapper_0_81.getLink(Database.java:595)
            at com.uphyca.stetho_realm.Database.flattenRows(Database.java:204)
            at com.uphyca.stetho_realm.Database.access$300(Database.java:37)
            at com.uphyca.stetho_realm.Database$1.handleSelect(Database.java:121)
            at com.uphyca.stetho_realm.Database$1.handleSelect(Database.java:101)
            at com.uphyca.stetho_realm.RealmPeerManager.executeSQL(RealmPeerManager.java:123)
            at com.uphyca.stetho_realm.Database.executeSQL(Database.java:100)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.facebook.stetho.inspector.MethodDispatcher$MethodDispatchHelper.invoke(MethodDispatcher.java:96)
            at com.facebook.stetho.inspector.MethodDispatcher.dispatch(MethodDispatcher.java:67)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteRequest(ChromeDevtoolsServer.java:129)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.handleRemoteMessage(ChromeDevtoolsServer.java:111)
            at com.facebook.stetho.inspector.ChromeDevtoolsServer.onMessage(ChromeDevtoolsServer.java:87)
            at com.facebook.stetho.websocket.WebSocketSession$1.handleTextFrame(WebSocketSession.java:176)
            at com.facebook.stetho.websocket.WebSocketSession$1.onCompleteFrame(WebSocketSession.java:136)
            at com.facebook.stetho.websocket.ReadHandler.readLoop(ReadHandler.java:46)
            at com.facebook.stetho.websocket.WebSocketSession.handle(WebSocketSession.java:45)
            at com.facebook.stetho.websocket.WebSocketHandler.doUpgrade(WebSocketHandler.java:154)
            at com.facebook.stetho.websocket.WebSocketHandler.handleSecured(WebSocketHandler.java:106)
            at com.facebook.stetho.server.SecureHttpRequestHandler.handle(SecureHttpRequestHandler.java:52)
            at org.apache.http.protocol.HttpService.doService(HttpService.java:248)
            at org.apache.http.protocol.HttpService.handleRequest(HttpService.java:192)
            at com.facebook.stetho.server.LocalSocketHttpServer$WorkerThread.run(LocalSocketHttpServer.java:233)

**Comment by skyisle** [comment]
Here is PR for this issue. https://github.com/uPhyca/stetho-realm/pull/5

**Comment by zaki50** [comment]
@bendywalker @pavel-mukhanov @skyisle thank you for your report & PR.

I'll check soon

**Comment by zaki50** [comment]
@marcelsanches2 Thank you for your report

**Comment by zaki50** [comment]
Released 0.7.2 which fix this issue.

If not fixed, please reopen this.
